File: frontend/src/api/case.ts
import api from './index'

export interface CaseItem {
  id: number
  title: string
  case_type: string
  severity: string
  scene_description: string
  recommended_actions: string
  process_info: string
  images: string[]
  source_record_id: number | null
  source_filename: string | null
  created_at: string
  updated_at: string
}

export interface CaseListData {
  items: CaseItem[]
  total: number
  page: number
  page_size: number
}

export interface CaseCreateData {
  title: string
  case_type: string
  severity: string
  scene_description?: string
  recommended_actions?: string
  process_info?: string
  images?: string[]
  source_record_id?: number | null
}

export function getCases(params: {
  page: number
  page_size: number
  case_type?: string
  severity?: string
  keyword?: string
}) {
  return api.get<CaseListData>('/cases', { params })
}

export function getCase(id: number) {
  return api.get<CaseItem>(`/cases/${id}`)
}

export function createCase(data: CaseCreateData) {
  return api.post<CaseItem>('/cases', data)
}

export function createCaseFromRecord(recordId: number) {
  return api.post<CaseItem>(`/cases/from-record/${recordId}`)
}

export function updateCase(id: number, data: Partial<CaseCreateData>) {
  return api.put<CaseItem>(`/cases/${id}`, data)
}

export function deleteCase(id: number) {
  return api.delete(`/cases/${id}`)
}

File: backend/app/schemas/case.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class CaseCreate(BaseModel):
    title: str
    case_type: str
    severity: str
    scene_description: str = ''
    recommended_actions: str = ''
    process_info: str = ''
    images: list[str] = []
    source_record_id: Optional[int] = None


class CaseUpdate(BaseModel):
    title: Optional[str] = None
    case_type: Optional[str] = None
    severity: Optional[str] = None
    scene_description: Optional[str] = None
    recommended_actions: Optional[str] = None
    process_info: Optional[str] = None
    images: Optional[list[str]] = None


class CaseResponse(BaseModel):
    id: int
    title: str
    case_type: str
    severity: str
    scene_description: str
    recommended_actions: str
    process_info: str
    images: list[str]
    source_record_id: Optional[int] = None
    source_filename: Optional[str] = None
    created_at: str
    updated_at: str


class CaseListResponse(BaseModel):
    items: list[CaseResponse]
    total: int
    page: int
    page_size: int
